azure/scope/clients.go

```diff
@@ -133,7 +140,12 @@ func (c *AzureClients) setCredentialsWithProvider(ctx context.Context, subscript
 	}
 	c.Values[auth.ClientSecret] = strings.TrimSuffix(clientSecret, "\n")
 
-	c.Authorizer, err = credentialsProvider.GetAuthorizer(ctx, c.ResourceManagerEndpoint, c.Environment.ActiveDirectoryEndpoint, c.Environment.TokenAudience)
+	tokenCredential, err := credentialsProvider.GetTokenCredential(ctx, c.ResourceManagerEndpoint, c.Environment.ActiveDirectoryEndpoint, c.Environment.TokenAudience)
```

Comment:
Should we add unit tests to `clients_test.go` for the `HashKey()` and `Token()` funcs?

Reply:
I think both those functions would have trivial unit tests, and `HashKey()` was implemented in a previous PR. I do agree that this file needs unit tests and I think we can implement tests for the other functions in a separate issue/PR. wdyt?